snapshot/pkg/cloudprovider/providers/openstack/openstack_snapshots.go

```diff
@@ -234,12 +234,17 @@ func (os *OpenStack) FindSnapshot(tags map[string]string) ([]string, []string, e
 				nameVal, ok := snapshot.Metadata[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotNameTag]
 				if ok {
 					if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotNameTag] == nameVal {
-						timeVal, ok := snapshot.Metadata[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotTimestampTag]
+						uidVal, ok := snapshot.Metadata[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotUIDTag]
 						if ok {
-							if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotTimestampTag] == timeVal {
-								snapshotIDs = append(snapshotIDs, snapshot.ID)
-								statuses = append(statuses, snapshot.Status)
-								glog.Infof("Add snapshot [%#v].", snapshot)
+							if tags[ctrlsnap.CloudSnapshotCreatedForVolumeSnapshotUIDTag] == uidVal {
```

Comment:
Does openstack support filter when listing snapshots? Use filter could avoid manually checking the meta data?

Reply:
Metadata based filtering is supported by Cinder volume API microversion 3.21 and above. Currently the OpenStack Cloud Provider only supports V2.0.  I'll add a note to revisit this later.

Reply:
A note is added.